[js] test order of host callout & typecheck during eval

mikesamuel opened the pull request:
This adds tests for tc39/ecma262#1504 which cannot be tested purely in
JavaScript because it involves HostEnsureCanCompileStrings.

As documented in a table on tc39/ecma262#1495 this passes on recent
Chrome and Safari and fails on Firefox so this test probably shouldn't
land right away.

mikesamuel committed:
[js] test order of host callout & typecheck during eval

This adds tests for tc39/ecma262#1504 which cannot be tested purely in
JavaScript because it involves HostEnsureCanCompileStrings.

As documented in a table on tc39/ecma262#1495 this passes on recent
Chrome and Safari and fails on Firefox so this test probably shouldn't
land right away.

jugglinmike commented:
Mind moving this to the `content-security-policy/` directory?

mikesamuel commented:
@jugglinmike I don't mind at all, but it's testing language in ECMA262, not language in CSP3.

jugglinmike commented:
[It's kind of defined in both places.](https://w3c.github.io/webappsec-csp/#ecma-integration) Storing the test in `content-security-policy` more clearly relates the functionality to the web platform, and I'm thinking it makes it more visible for CSP's editors.

Alternatively, you might consider testing this in Test262 via some new host-defined behavior along [the same lines as `IsHTMLDDA`/`document.all`](https://github.com/tc39/test262/pull/1305). I haven't review the semantics, though, so it might not be a good match for this.

mikesamuel commented:
@jugglinmike Will move it over.

mikesamuel committed:
Moved test to content-security-policy/generic

I reformatted it to follow the guidance at

https://github.com/web-platform-tests/wpt/blob/38f02ed8bb05f69ab4b061d17220f8f46709a0a3/content-security-policy/README.html#L37-L60

and moved the `<meta http-equiv=Content-Security-Policy>` content into a separate file per

https://github.com/web-platform-tests/wpt/blob/38f02ed8bb05f69ab4b061d17220f8f46709a0a3/content-security-policy/README.html#L22-L24

mikesamuel commented:
@jugglinmike Moved and reworked to follow conventions local to that directory.

jugglinmike commented on content-security-policy/generic/eval-typecheck-callout-order.html:
```diff
@@ -0,0 +1,28 @@
+    <script src='/resources/testharnessreport.js' nonce='abc'></script>
+</head>
+<body>
+    <title>Test for order of Type(evalInput) and host callout</title>
```
Did you mean for this to go in the `<head>`?

mikesamuel replied:
Yes.  Fixed.

jugglinmike commented:
> @jugglinmike Moved and reworked to follow conventions local to that directory.

Thanks!

> As documented in a table on tc39/ecma262#1495 this passes on recent Chrome and Safari and fails on Firefox so this test probably shouldn't land right away.

We can add "do not merge yet" to the pull request, or we could merge it with the ".tentative" file name marker. The latter is less likely to be forgotten, so if this isn't contentious, then that might be best. @mikewest do you have a preference?

mikesamuel committed:
move `<title>` into `<head>`

mikewest commented:
The test seems accurate to me, assuming https://github.com/tc39/ecma262/pull/1504 lands. I'd be fine with y'all landing these tests now as `.tentative`, or waiting until that patch lands in ecmascript and landing the patches without that designation. Either way is great!

mikesamuel committed:
Mark new test tentative

... until tc39/ecma262#1495 lands.

mikesamuel commented:
The test is now tentative.

```sh
$ git mv content-security-policy/generic/eval-typecheck-callout-order{,.tentative}.html
$ git mv content-security-policy/generic/eval-typecheck-callout-order{,.tentative}.html.headers
```

jugglinmike reviewed (approved):
Looks good to me
